include/core/Layer.hpp

```diff
@@ -64,6 +83,19 @@ namespace ngen
 
         virtual ~Layer() {}
 
+        /**
+        * @brief Get a class id for this layer object
+        * 
+        * @return int
+        */
+        virtual int class_id() { return LayerClass::kLayer; }
+
+        /**
+         * @brief Get a list of output variables names for this layer
+         * 
+         * @return vector of output variable names
+        */
+
```

Comment:
Did you __ a function here?

Reply:
I think that was a function that was removed as redundant.

Reply:
The actual information comes from the BMI realizations of the contained models, it would be nice for the layer to have a summary but currently it is not needed to get things to work. This could be used to limit what variables are output.